// @capawesome-team/capacitor-contacts/dist/esm/definitions.js
/**
 * @since 7.0.0
 */
export var EmailAddressType;
(function (EmailAddressType) {
    /**
     * @since 7.0.0
     */
    EmailAddressType["Custom"] = "CUSTOM";
    /**
     * @since 7.0.0
     */
    EmailAddressType["Home"] = "HOME";
    /**
     * Only available on iOS.
     *
     * @since 7.0.0
     */
    EmailAddressType["ICloud"] = "ICLOUD";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    EmailAddressType["Mobile"] = "MOBILE";
    /**
     * @since 7.0.0
     */
    EmailAddressType["Other"] = "OTHER";
    /**
     * Only available on iOS.
     *
     * @since 7.0.0
     */
    EmailAddressType["School"] = "SCHOOL";
    /**
     * @since 7.0.0
     */
    EmailAddressType["Work"] = "WORK";
})(EmailAddressType || (EmailAddressType = {}));
/**
 * @since 7.0.0
 */
export var PhoneNumberType;
(function (PhoneNumberType) {
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["Assistant"] = "ASSISTANT";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["Callback"] = "CALLBACK";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["Car"] = "CAR";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["CompanyMain"] = "COMPANY_MAIN";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["Custom"] = "CUSTOM";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["FaxHome"] = "FAX_HOME";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["FaxOther"] = "FAX_OTHER";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["FaxWork"] = "FAX_WORK";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["Home"] = "HOME";
    /**
     * Only available on iOS.
     *
     * @since 7.0.0
     */
    PhoneNumberType["IPhone"] = "IPHONE";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["Isdn"] = "ISDN";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["Main"] = "MAIN";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["Mms"] = "MMS";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["Mobile"] = "MOBILE";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["Other"] = "OTHER";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["Pager"] = "PAGER";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["Radio"] = "RADIO";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["Telex"] = "TELEX";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["TtyTdd"] = "TTY_TDD";
    /**
     * @since 7.0.0
     */
    PhoneNumberType["Work"] = "WORK";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["WorkMobile"] = "WORK_MOBILE";
    /**
     * Only available on Android.
     *
     * @since 7.0.0
     */
    PhoneNumberType["WorkPager"] = "WORK_PAGER";
})(PhoneNumberType || (PhoneNumberType = {}));
/**
 * @since 7.0.0
 */
export var PostalAddressType;
(function (PostalAddressType) {
    /**
     * @since 7.0.0
     */
    PostalAddressType["Custom"] = "CUSTOM";
    /**
     * @since 7.0.0
     */
    PostalAddressType["Home"] = "HOME";
    /**
     * @since 7.0.0
     */
    PostalAddressType["Other"] = "OTHER";
    /**
     * @since 7.0.0
     */
    PostalAddressType["Work"] = "WORK";
})(PostalAddressType || (PostalAddressType = {}));
export var UrlAddressType;
(function (UrlAddressType) {
    /**
     * Only available on Android.
     *
     * @since 7.5.0
     */
    UrlAddressType["Blog"] = "BLOG";
    /**
     * @since 7.5.0
     */
    UrlAddressType["Custom"] = "CUSTOM";
    /**
     * Only available on Android.
     *
     * @since 7.5.0
     */
    UrlAddressType["Ftp"] = "FTP";
    /**
     * @since 7.5.0
     */
    UrlAddressType["Home"] = "HOME";
    /**
     * @since 7.5.0
     */
    UrlAddressType["Homepage"] = "HOMEPAGE";
    /**
     * @since 7.5.0
     */
    UrlAddressType["Other"] = "OTHER";
    /**
     * Only available on Android.
     *
     * @since 7.5.0
     */
    UrlAddressType["Profile"] = "PROFILE";
    /**
     * Only available on iOS.
     *
     * @since 7.5.0
     */
    UrlAddressType["School"] = "SCHOOL";
    /**
     * @since 7.5.0
     */
    UrlAddressType["Work"] = "WORK";
})(UrlAddressType || (UrlAddressType = {}));
